docs/IcepeakProtocolMultiSub.md

```diff
@@ -0,0 +1,177 @@
+# The Icepeak Multiple-Subcription-Single-Connection Protocol
+
+This protocol was developed atop the existing protocol to allow the client to subscribe to multiple "paths"/"topics" using a single websocket connection.
+
+Whereas the original existing protocol required a new websocket connection for each path the client wanted to subscribe to.
+
+# Initial Connection
+
+A client can connect to a multiple-subscription protocol connection with a URL such as:
+`ws://yourdomain.com/?method=reusable`
+
+The URL path:
+- points to the root of your icepeak service
+- has the query parameter `method` set to `reusable`
+
+The request should contain:
+- a token of autherisation about the right to establish a connection with the server **UNDER-SPECIFIED** 
+
+# Subscribing, Unsubscribing & Updates
+
+In summary:
+- The client can request to subscribe to an array of paths.
+- The client, on the single connection, can cummulavely keep subscribing and unsubscribing to paths by sending corresponding payloads.
+  - The server also sends back a response about the status, and some data of the corresponding subscription or unsubscription request.
+- The server will send the client the new value/update at the subscribed path whenever there is a change at that path.
+- Both, subscription and unsubscription requests are idempotent.
+
+## Update
+
+`JSON Schema` declaration of the update the server will send to the client upon subscribed path change:
+```javascript
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "wss://updates.channable.com",
+  "title": "Path changes",
+  "description": "Indicates changes occurred at the subscribed paths",
+  "type": "object",
+  "properties": {
+    "type": { "const": "update" },
+    "change": {
+      "type": "object",
+      "properties": {
+        "path": { "type": "string" },
+        "value": {}
+      }
+    }
+  }
+}
+```
+
+## Subscribe
+
+In summary:
+- The client can send a payload that contains an array of paths to subscribe to.
+  - The client has to send a JWT that authorises all the paths.
+- The client can expect a response from the server that contains the status/acknowledgement of the request.
+  - Upon a successful request, the client can expect the payload to also contain the current value of the paths requested.
+  
```

Comment:
What do you mean? The whitespace before `-` are intended.

Reply:
I meant the whitespace on the otherwise empty line. There's quite a lot of it throughout the PR. I don't think Github shows whitespaces in diffs, so this isn't visible in the webinterface.

Reply:
Ah I see. Configured my editor to show trailing whitespace. Nice